README.md
[![Build Status](https://travis-ci.org/SimplySecurity/SimplyEmail.svg?branch=master)](https://travis-ci.org/SimplySecurity/SimplyEmail)
[![Coverage Status](https://coveralls.io/repos/github/killswitch-GUI/SimplyEmail/badge.svg?branch=Version-1.4)](https://coveralls.io/github/killswitch-GUI/SimplyEmail?branch=Version-1.4)

![alt text](https://simplyemail.org/img/se-logo-2.png "Logo Title Text 1")

----
SimplyEmail was built arround the concept that tools should do somthing, and do that somthing well, hence 'Simply'. Full documentation can be found at:

[INSTALL / FAQ] https://simplysecurity.github.io/SimplyEmail/

[CHANGELOG] https://github.com/SimplySecurity/SimplyEmail/blob/master/CHANGELOG.md

[HELP/QUESTIONS/CHAT] Join us at: https://simplysecurity.herokuapp.com

## TL;DR 
Supported Platforms / Tested with CI:
* **Docker**
* Kali Rolling
* Debian 9.x
* Ubuntu 16.04 LTS
* MacOS


```bash
root@kali:~# curl -s https://raw.githubusercontent.com/killswitch-GUI/SimplyEmail/master/setup/oneline-setup.sh | bash
root@kali:~# cd SimplyEmail
(SE) root@kali:~/SimplyEmail# ./SimplyEmail.py
```

or
 
```bash
root@kali:~# docker pull simplysecurity/simplyemail
root@kali:~# docker run -ti simplysecurity/simplyemail
```


CHANGELOG.md
# Change Log

## [Version-1.4.10](https://github.com/SimplySecurity/SimplyEmail/tree/Version-1.4.10) (2017-12-22)
[Full Changelog](https://github.com/SimplySecurity/SimplyEmail/compare/Version-1.4.9...Version-1.4.10)

**Implemented enhancements:**

- Overhaul EmailHunter Module [\#34](https://github.com/SimplySecurity/SimplyEmail/pull/34) ([capnspacehook](https://github.com/capnspacehook))

**Closed issues:**

- Modules not installed [\#41](https://github.com/SimplySecurity/SimplyEmail/issues/41)

## [Version-1.4.9](https://github.com/SimplySecurity/SimplyEmail/tree/Version-1.4.9) (2017-12-22)
[Full Changelog](https://github.com/SimplySecurity/SimplyEmail/compare/Version-1.4.8...Version-1.4.9)

## [Version-1.4.8](https://github.com/SimplySecurity/SimplyEmail/tree/Version-1.4.8) (2017-12-22)
[Full Changelog](https://github.com/SimplySecurity/SimplyEmail/compare/Version-1.4.7...Version-1.4.8)

## [Version-1.4.7](https://github.com/SimplySecurity/SimplyEmail/tree/Version-1.4.7) (2017-12-21)
[Full Changelog](https://github.com/SimplySecurity/SimplyEmail/compare/Version-1.4.6...Version-1.4.7)

## [Version-1.4.6](https://github.com/SimplySecurity/SimplyEmail/tree/Version-1.4.6) (2017-12-21)
[Full Changelog](https://github.com/SimplySecurity/SimplyEmail/compare/Version-1.4.5...Version-1.4.6)

## [Version-1.4.5](https://github.com/SimplySecurity/SimplyEmail/tree/Version-1.4.5) (2017-12-21)
[Full Changelog](https://github.com/SimplySecurity/SimplyEmail/compare/V1.4.1...Version-1.4.5)

**Implemented enhancements:**

- Module sort [\#17](https://github.com/SimplySecurity/SimplyEmail/issues/17)

**Fixed bugs:**

- AttributeError: 'module' object has no attribute '\_base' [\#29](https://github.com/SimplySecurity/SimplyEmail/issues/29)
- Installation script brakes Kali  [\#24](https://github.com/SimplySecurity/SimplyEmail/issues/24)
- Execution Error [\#14](https://github.com/SimplySecurity/SimplyEmail/issues/14)

**Closed issues:**

- Magic not working on KALI latest [\#35](https://github.com/SimplySecurity/SimplyEmail/issues/35)
- canar.io [\#31](https://github.com/SimplySecurity/SimplyEmail/issues/31)
- Easier automation [\#15](https://github.com/SimplySecurity/SimplyEmail/issues/15)

**Merged pull requests:**

- test [\#43](https://github.com/SimplySecurity/SimplyEmail/pull/43) ([killswitch-GUI](https://github.com/killswitch-GUI))
- Requirements file has double requirements [\#42](https://github.com/SimplySecurity/SimplyEmail/pull/42) ([Pysint](https://github.com/Pysint))
- Create CODE\_OF\_CONDUCT.md [\#39](https://github.com/SimplySecurity/SimplyEmail/pull/39) ([killswitch-GUI](https://github.com/killswitch-GUI))
- Create LICENSE [\#38](https://github.com/SimplySecurity/SimplyEmail/pull/38) ([killswitch-GUI](https://github.com/killswitch-GUI))
- Update requirments.txt [\#37](https://github.com/SimplySecurity/SimplyEmail/pull/37) ([killswitch-GUI](https://github.com/killswitch-GUI))
- Dev [\#36](https://github.com/SimplySecurity/SimplyEmail/pull/36) ([killswitch-GUI](https://github.com/killswitch-GUI))
- Add `--upgrade` flag [\#30](https://github.com/SimplySecurity/SimplyEmail/pull/30) ([bcoles](https://github.com/bcoles))
- OS system check for Setup.sh [\#27](https://github.com/SimplySecurity/SimplyEmail/pull/27) ([y4utj4](https://github.com/y4utj4))
- Fix wrong bracket [\#26](https://github.com/SimplySecurity/SimplyEmail/pull/26) ([DPeltier](https://github.com/DPeltier))
- Fix wget errors [\#25](https://github.com/SimplySecurity/SimplyEmail/pull/25) ([DPeltier](https://github.com/DPeltier))
- Fixed chmod trying to mod file at wrong location [\#23](https://github.com/SimplySecurity/SimplyEmail/pull/23) ([ChrisTruncer](https://github.com/ChrisTruncer))
- Dev sleep/jitter [\#22](https://github.com/SimplySecurity/SimplyEmail/pull/22) ([killswitch-GUI](https://github.com/killswitch-GUI))
- Dev json [\#20](https://github.com/SimplySecurity/SimplyEmail/pull/20) ([killswitch-GUI](https://github.com/killswitch-GUI))
- Add ordered list [\#19](https://github.com/SimplySecurity/SimplyEmail/pull/19) ([killswitch-GUI](https://github.com/killswitch-GUI))

## [V1.4.1](https://github.com/SimplySecurity/SimplyEmail/tree/V1.4.1) (2016-06-20)
[Full Changelog](https://github.com/SimplySecurity/SimplyEmail/compare/v1.3...V1.4.1)

**Implemented enhancements:**

- Please use the API [\#9](https://github.com/SimplySecurity/SimplyEmail/issues/9)
- A bunch of spellcheck/pep8 stuff, and some small improvements [\#13](https://github.com/SimplySecurity/SimplyEmail/pull/13) ([rvazarkar](https://github.com/rvazarkar))

**Fixed bugs:**

- Error in File Location:  [\#16](https://github.com/SimplySecurity/SimplyEmail/issues/16)
- Error, Kali 2016 Rolling Release [\#11](https://github.com/SimplySecurity/SimplyEmail/issues/11)
- Please use the API [\#9](https://github.com/SimplySecurity/SimplyEmail/issues/9)

**Merged pull requests:**

- Dev domain [\#18](https://github.com/SimplySecurity/SimplyEmail/pull/18) ([killswitch-GUI](https://github.com/killswitch-GUI))
- Version 1.4 [\#12](https://github.com/SimplySecurity/SimplyEmail/pull/12) ([killswitch-GUI](https://github.com/killswitch-GUI))

## [v1.3](https://github.com/SimplySecurity/SimplyEmail/tree/v1.3) (2016-04-10)
[Full Changelog](https://github.com/SimplySecurity/SimplyEmail/compare/1.2...v1.3)

## [1.2](https://github.com/SimplySecurity/SimplyEmail/tree/1.2) (2016-02-20)
[Full Changelog](https://github.com/SimplySecurity/SimplyEmail/compare/1.0...1.2)

## [1.0](https://github.com/SimplySecurity/SimplyEmail/tree/1.0) (2016-02-06)
[Full Changelog](https://github.com/SimplySecurity/SimplyEmail/compare/v0.9...1.0)

**Implemented enhancements:**

- fixing some errors and Add Exalead Searches [\#8](https://github.com/SimplySecurity/SimplyEmail/pull/8) ([th3gundy](https://github.com/th3gundy))

## [v0.9](https://github.com/SimplySecurity/SimplyEmail/tree/v0.9) (2016-02-01)
[Full Changelog](https://github.com/SimplySecurity/SimplyEmail/compare/0.8...v0.9)


Helpers/helpers.py
#!/usr/bin/env python

import os
import textwrap
import logging
import time
import magic
import json
import configparser
import collections
import random
from fake_useragent import UserAgent

def dictToJson(inputDict):
    """
    Takes in a list of dict items.
    Converts them to json and returns list of json obj.
    """
    obj = []
    for item in inputDict:
        obj += json.dumps(item)
    return obj

def get_searchid():
    currentDate = str(time.strftime("%d%m%Y"))
    currentTime = str(time.strftime("%H%M%S"))
    searchid = currentDate + currentTime
    return searchid

def get_datetime():
    currentDate = str(time.strftime("%d/%m/%Y"))
    currentTime = str(time.strftime("%H:%M:%S"))
    datetime = currentDate + ' ' +currentTime
    return datetime

def JsonListToJsonObj(inputJsonList, domain):
    """
    Takes a list of json objects,
    places them in a key and returns the data.
    """
    currentDate = str(time.strftime("%d/%m/%Y"))
    currentTime = str(time.strftime("%H:%M:%S"))
    currentTool = "SimplyEmail"
    config = configparser.ConfigParser()
    config.read('Common/SimplyEmail.ini')
    currentVersion = str(config['GlobalSettings']['Version'])
    count = len(inputJsonList)
    dic = collections.OrderedDict()
    dic = {
        "domain_of_collection" : domain,
        "data_of_collection" : currentDate,
        "time_of_collection" : currentTime,
        "tool_of_collection" : currentTool,
        "current_version" :  currentVersion,
        "email_collection_count" : count,
        "emails" : inputJsonList,
    }
    obj = json.dumps(dic, indent=4, sort_keys=True)
    return obj

def color(string, status=True, warning=False, bold=True, blue=False, firewall=False):
    # Change text color for the linux terminal, defaults to green.
    # Set "warning=True" for red.
    # stolen from Veil :)
    attr = []
    if status:
        # green
        attr.append('32')
    if warning:
        # red
        attr.append('31')
    if bold:
        attr.append('1')
    if firewall:
        attr.append('33')
    if blue:
        # blue
        attr.append('34')
    return '\x1b[%sm%s\x1b[0m' % (';'.join(attr), string)


def formatLong(title, message, frontTab=True, spacing=16):
    """
    Print a long title:message with our standardized formatting.
    Wraps multiple lines into a nice paragraph format.
    """

    lines = textwrap.wrap(textwrap.dedent(message).strip(), width=50)
    returnstring = ""

    i = 1
    if len(lines) > 0:
        if frontTab:
            returnstring += "\t%s%s" % (('{0: <%s}' %
                                         spacing).format(title), lines[0])
        else:
            returnstring += " %s%s" % (('{0: <%s}' %
                                        (spacing-1)).format(title), lines[0])
    while i < len(lines):
        if frontTab:
            returnstring += "\n\t"+' '*spacing+lines[i]
        else:
            returnstring += "\n"+' '*spacing+lines[i]
        i += 1
    return returnstring


def DirectoryListing(directory):
    # Returns a list of dir's of results
    dirs = []
    for (dir, _, files) in os.walk(directory):
        for f in files:
            path = os.path.join(dir, f)
            if os.path.exists(path):
                dirs.append(path)
    return dirs

def split_email(email):
    email = email.lower()
    se = email.split("@")
    return se

def getua():
    # gets a random useragent and returns the UA
    ua = UserAgent()
    return ua.random

def modsleep(delay, jitter=0):
    # Quick Snipit From EmPyre Agent (@HarmJ0y)
    if jitter < 0: jitter = -jitter
    if jitter > 1: jitter = 1/jitter

    minSleep = int((1.0-jitter)*delay)
    maxSleep = int((1.0+jitter)*delay)
    sleepTime = random.randint(minSleep, maxSleep)
    time.sleep(int(sleepTime))

def filetype(path):
    m = magic.from_file(str(path))
    return m

#######################
# Setup Logging Class #
#######################


class log(object):

    """simple logging testing and dev"""

    def __init__(self):
        self.name = ".SimplyEmail.log"

    def start(self):
        logger = logging.getLogger("SimplyEmail")
        logger.setLevel(logging.INFO)
        fh = logging.FileHandler(self.name)
        formatter = logging.Formatter(
            '%(asctime)s-[%(name)s]-[%(levelname)s]- %(message)s')
        fh.setFormatter(formatter)
        logger.addHandler(fh)
        logger.info("Program started")
        logging.captureWarnings(True)
        logger.info("Set Logging Warning Capture: True")

    def infomsg(self, message, modulename):
        try:
            msg = 'SimplyEmail.' + str(modulename)
            logger = logging.getLogger(msg)
            logger.info(str(message))
        except Exception as e:
            print e

    def warningmsg(self, message, modulename):
        try:
            msg = 'SimplyEmail.' + str(modulename)
            logger = logging.getLogger(msg)
            logger.warning(str(message))
        except Exception as e:
            print e
